File: frontend/src/lib/dtos/forest-stand/forest-stand.dto.ts
export type RecentActivityDto = {
	id: string;
	description: string;
	quantity: number;
	unit: string | null;
	date: string;
	activityTypeName: string;
	userName: string;
	cadasterCadastralNumber: string | null;
	forestStandNumber: number;
};

export type ActivityListDto = {
	id: string;
	description: string;
	quantity: number;
	unit: string | null;
	date: string;
	activityTypeName: string;
	userName: string;
	cadasterCadastralNumber: string | null;
	forestStandNumber: number;
	locationDescription: string | null;
	applicationStatus: string | null;
};

export type ForestStandDto = {
	id: string;
	number: number;
	area: number;
	totalVolume: number;
	isActive: boolean;
	validFrom: string;
	validTo: string | null;
	cadasterId: string;
	cadasterCadastralNumber: string;
	landPropertyId: string;
	landPropertyName: string;
	landPropertyIsFsc: boolean;
	recentActivities: RecentActivityDto[];
};

export type ForestStandSummaryDto = {
	id: string;
	number: number;
	cadasterId: string;
	cadasterCadastralNumber: string;
	landPropertyId?: string;
	landPropertyName?: string;
};

export type ForestStandUpdateDto = {
	id: string;
	number: number;
	area: number;
	totalVolume: number;
	isActive: boolean;
	validFrom: string;
	validTo: string | null;
	cadasterId: string;
};

export type CadasterSummaryDto = {
	id: string;
	cadastralNumber: string;
	landPropertyId: string;
	landPropertyName: string;
	landPropertyIsFsc: boolean;
};

File: ForestManagement/App.DTO/ForestStand/ForestStandDto.cs
using App.DTO.Activity;

namespace App.DTO.ForestStand;

/// <summary>
/// Full forest stand details response DTO
/// </summary>
public class ForestStandDto
{
    public Guid Id { get; set; }
    public int Number { get; set; }
    public decimal Area { get; set; }
    public int TotalVolume { get; set; }
    public bool IsActive { get; set; }
    public DateTime ValidFrom { get; set; }
    public DateTime? ValidTo { get; set; }
    public Guid CadasterId { get; set; }
    public string CadasterCadastralNumber { get; set; } = string.Empty;
    public Guid LandPropertyId { get; set; }
    public string LandPropertyName { get; set; } = string.Empty;
    public bool LandPropertyIsFsc { get; set; }
    
    // Recent activities (5 most recent)
    public ICollection<RecentActivityDto> RecentActivities { get; set; } = new List<RecentActivityDto>();
}
